### PR TITLE
Fix DuckDuckGo HTML search

### LINKED ISSUE #22090: DuckDuckGo search results in "forbidden"
URL: https://duckduckgo.com/html

Steps to reproduce:

1. Load https://duckduckgo.com/html
2. Search something.
3. The search result page does not appear but instead the text "forbidden" is displayed.

This issue first occurred in the [2018-11-02](http://servo-builds.s3.amazonaws.com/) nightly build. Tested in Windows 10.

### PR DESCRIPTION
**Pull request by jdm**
- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] These changes fix #22090
- [x] There are tests for these changes

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/22354)

### CODE REVIEW

**Comment by jdm** [comment]
I can't test the Origin header change meaningfully, unfortunately, since the WPT tests run on ports 8000 and 8443 by default.

File: tests/wpt/web-platform-tests/common/form-submission.py [hunk 1]
```diff
@@ -6,5 +6,7 @@ def main(request, response):
     else:
         result = request.POST.first('foo') == 'bar'
 
+    results = result and request.url_parts.query == 'query=1'
```
Comment (Manishearth) [inline tests/wpt/web-platform-tests/common/form-submission.py]: `results` isn't used at all here

**Review by Manishearth — APPROVED** [review]

File: tests/wpt/web-platform-tests/common/form-submission.py (hunk 1 above)
Comment (jdm) [reply tests/wpt/web-platform-tests/common/form-submission.py]: Aieee.

**Comment by CYBAI** [comment]
```diff
Diff in /home/travis/build/servo/servo/components/net/tests/http_loader.rs at line 1450:
     );
 
     let origin_url = ServoUrl::parse("http://example.com").unwrap();
-    origin = Origin::try_from_parts(
-        origin_url.scheme(),
-        origin_url.host_str().unwrap(),
-        None,
-    )
-    .unwrap();
+    origin =
+        Origin::try_from_parts(origin_url.scheme(), origin_url.host_str().unwrap(), None).unwrap();
     // Test Origin header is set on Get request with CORS mode
     let mut request = Request::from_init(RequestInit {
         url: url.clone(),
Run `./mach fmt` to fix the formatting
```

**Comment by CYBAI** [comment]
@bors-servo retry

- #22271

**Comment by CYBAI** [comment]
@bors-servo retry

```
{
    "status": "CRASH", 
    "group": "default", 
    "message": null, 
    "stack": null, 
    "subtest": null, 
    "test": "/html/browsers/browsing-the-web/navigating-across-documents/navigation_unload_data_url.html", 
    "line": 98554, 
    "action": "test_result", 
    "expected": "OK"
}
{
    "status": "TIMEOUT", 
    "group": "default", 
    "message": null, 
    "stack": null, 
    "subtest": null, 
    "test": "/webstorage/storage_enumerate.html", 
    "line": 132098, 
    "action": "test_result", 
    "expected": "OK"
}
{
    "status": "TIMEOUT", 
    "group": "default", 
    "message": null, 
    "stack": null, 
    "subtest": null, 
    "test": "/domparsing/insert-adjacent.html", 
    "line": 132147, 
    "action": "test_result", 
    "expected": "OK"
}
{
    "status": "TIMEOUT", 
    "group": "default", 
    "message": null, 
    "stack": null, 
    "subtest": null, 
    "test": "/dom/nodes/getElementsByClassName-11.xml", 
    "line": 132161, 
    "action": "test_result", 
    "expected": "ERROR"
}
```

**Comment by CYBAI** [comment]
@bors-servo retry

- Windows oom

**Comment by CYBAI** [comment]
@bors-servo retry

```
{
    "status": "FAIL", 
    "group": "default", 
    "message": "assert_true: expected true got false", 
    "stack": "source.onopen/<@http://web-platform.test:8000/eventsource/format-field-retry.htm:23:15\nTest.prototype.step@http://web-platform.test:8000/resources/testharness.js:1566:20\nsource.onopen@http://web-platform.test:8000/eventsource/format-field-retry.htm:18:11\n", 
    "subtest": "EventSource: \"retry\" field", 
    "test": "/eventsource/format-field-retry.htm", 
    "line": 15161, 
    "action": "test_result", 
    "expected": "PASS"
}
```

**Comment by CYBAI** [comment]
@bors-servo retry

```
{
    "status": "TIMEOUT", 
    "group": "default", 
    "message": null, 
    "stack": null, 
    "subtest": null, 
    "test": "/workers/semantics/interface-objects/003.html", 
    "line": 143882, 
    "action": "test_result", 
    "expected": "ERROR"
}
```

**Comment by CYBAI** [comment]
@bors-servo retry

```
{
    "status": "CRASH", 
    "group": "default", 
    "message": null, 
    "stack": null, 
    "subtest": null, 
    "test": "/html/syntax/parsing/html5lib_menuitem-element.html?run_type=uri", 
    "line": 134340, 
    "action": "test_result", 
    "expected": "OK"
}
```

**Comment by CYBAI** [comment]
@bors-servo retry

- #22492

**Comment by jdm** [comment]
@bors-servo r-
Emptying the queue so #22493 can merge easily.

**Pull request merged**